docs/editor/testing.md

```diff
@@ -0,0 +1,173 @@
+
+In this article, you'll learn how to get started with testing in VS Code, discover popular testing extensions, and explore the testing capabilities.
+
+![Testing in Visual Studio Code](images/testing/testing-hero.png)
+
+## Getting started with testing in VS Code
+
+Testing support in VS Code is language-specific and depends on the [extensions](#testing-extensions) you have installed. The testing features are provided by the language extensions, which can be language servers, test runners, or other tools that support testing in that language. To get started with testing in VS Code, first install the appropriate language extension for your project.
+
+Once you have the language extension installed, you can start discovering and running tests in your project. The [Test Explorer](#automatic-test-discovery-in-test-explorer) provides a centralized place to manage and run your tests.
+
+After you run your tests, you can view the test results in the Test Explorer view, editor gutter, and Test Results panel. To diagnose issues with your tests, you can also [run and debug tests](#run-and-debug-tests) and set breakpoints in your test and application code.
+
+You can also [run tests with coverage](#test-coverage) to see how much of your code is covered by your tests. Test coverage helps you identify areas of your code that are not being tested and ensures that your tests are comprehensive. You can view the test coverage results in different views in the editor and Test Coverage panel.
+
+To optimize your testing workflow, you can [create tasks to run your tests](#task-integration), and optionally run your tests in the background with every code change.
+
+## Testing extensions
+
+You can find extensions that support testing by looking in the [Visual Studio Marketplace](https://marketplace.visualstudio.com/search?target=VSCode&category=Testing&sortBy=Installs). You can also go to the Extensions view (`kb(workbench.view.extensions)`), and filter by the **Testing** category.
+
+<!-- <div class="marketplace-extensions-testing"></div> -->
+
+> **Tip**: The extensions shown above are dynamically queried. Select an extension tile to read the description and reviews to decide which extension is best for you.
+
+## Automatic test discovery in Test Explorer
+
+The Test Explorer view provides a centralized place to manage and run your tests. You can access the Test Explorer view by selecting the beaker icon in the Activity Bar or by using the **Testing: Focus on Test Explorer View** command in the Command Palette (`kb(workbench.action.showCommands)`).
+
+If you have a project with tests, the Test Explorer view automatically discovers and lists the tests in your workspace. By default, the discovered tests are displayed in a tree view in the Test Explorer. The tree view matches the hierarchical structure of your tests, making it easy to navigate and run your tests.
```

Comment:
There's also a little bit of "it depends" here, for example the Python extension requires users to run a configure command before it shows tests. (A button to do so is shown in the welcome view if there are no tests in the workspace yet.) The user might need to consult the readme of their extension to set up tests.

Reply:
Added a note at the end of this section.